src/backends/pyplot.jl

```diff
@@ -937,13 +937,21 @@ function _before_layout_calcs(plt::Plot{PyPlotBackend})
                 kw[:boundaries] = vcat(0, kw[:values] + 0.5)
             elseif any(colorbar_series[attr] !== nothing for attr in (:line_z, :fill_z, :marker_z))
                 cmin, cmax = get_clims(sp)
-                norm = pycolors."Normalize"(vmin = cmin, vmax = cmax)
+                norm = if sp[:colorbar_scale] === :identity
```

Comment:
This makes the matplotlib draw the colorbar properly (or the color gradient strip does not look linear)